feat(curriculum): add bisection method project to Python (#54110)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/07-scientific-computing-with-python/learn-numerical-methods-by-finding-the-square-root-of-a-number-using-the-bisection-method/65ef1d924ff2e5c7031c6ce8.md

---
id: 65ef1d924ff2e5c7031c6ce8
title: Step 18
challengeType: 20
dashedName: step-18
---

# --description--

> Create an `else` condition to handle the case where the value of `root` is not `None` and found as a result of convergance. If it is not `None`, print the message `'The approximate square root of {square_target} is {root}.'`. Remember to format using an f-string.

Reviewer: ```suggestion
Create an `else` clause to handle the case where the value of `root` is not `None` and found as a result of convergence. If it is not `None`, print the message `'The square root of {square_target} is approximately {root}'`. Remember to format the message using an f-string.
```

Reviewer: Maybe it's just me, but "found as a result of convergence" does not sound right.

Author: I modified the instruction a bit.

> Create an `else` clause to handle the case where the value of `root` is not `None`, indicating that a root hass been found. If it is not `None`, print the message `'The approximate square root of {square_target} is {root}.'`. Remember to format using an f-string.

Reviewer: ```suggestion
Create an `else` clause to handle the case where the value of `root` is not `None`, indicating that a root hass been found. If it is not `None`, print the message `'The square root of {square_target} is approximately {root}'`. Remember to format using an f-string.
```

# --hints--

> There should be an `else` condition

Reviewer: ```suggestion
You should create an `else` clause.
```

```js
({
    test: () => 
    {
         assert(runPython(`_Node(_code).find_function("square_root_bisection").find_ifs()[1].find_bodies()[2].find_ifs()[0].find_conditions()[1] == _Node()`));
    }
})

```

> The body of `else` should print 

Reviewer: ```suggestion
You should print `f'The square root of {square_target} is approximately {root}'` within the `else` body.
```
